docs/source/API/core/policies/RangePolicy.rst

```diff
@@ -120,27 +120,28 @@ Constructors
 
    Default Constructor uninitialized policy.
 
-.. cppkokkos:function:: RangePolicy(int64_t begin, int64_t end)
+.. cppkokkos:function:: RangePolicy(index_type begin, index_type end)
 
-   Provide a start and end index.
+   Provide a begin and end index.
 
-.. cppkokkos:function:: RangePolicy(int64_t begin, int64_t end, ChunkSize chunk_size)
+.. cppkokkos:function:: RangePolicy(index_type begin, index_type end, ChunkSize chunk_size)
 
-   Provide a start and end index as well as a ``ChunkSize``.
+   Provide a begin and end index as well as a ``ChunkSize``.
 
-.. cppkokkos:function:: RangePolicy(const ExecutionSpace& space, int64_t begin, int64_t end)
+.. cppkokkos:function:: RangePolicy(const ExecutionSpace& space, index_type begin, index_type end)
 
-   Provide a start and end index and an ``ExecutionSpace`` instance to use as the execution resource.
+   Provide a begin and end index and an ``ExecutionSpace`` instance to use as the execution resource.
 
-.. cppkokkos:function:: RangePolicy(const ExecutionSpace& space, int64_t begin, int64_t end, ChunkSize chunk_size)
+.. cppkokkos:function:: RangePolicy(const ExecutionSpace& space, index_type begin, index_type end, ChunkSize chunk_size)
 
-   Provide a start and end index and an ``ExecutionSpace`` instance to use as the execution resource, as well as a ``ChunkSize``.
+   Provide a begin and end index and an ``ExecutionSpace`` instance to use as the execution resource, as well as a ``ChunkSize``.
 
 Preconditions:
 ^^^^^^^^^^^^^^
 
-* The start index must not be greater than the end index.
+* The begin index must not be greater than the end index.
 * The actual constructors are templated so we can check that they are converted to ``index_type`` safely (see `#6754 <https://github.com/kokkos/kokkos/pull/6754>`_).
+   * The conversion safety check is performed only if ``index_type`` is convertible to the begin and end index types.
```

Comment:
```suggestion
* The actual constructors are templated so we can check that they are converted to ``index_type`` safely if possible (see `#6754 <https://github.com/kokkos/kokkos/pull/6754>`_).
```

Line 144 doesn't work, for the following reasons:

- You have to mention the template types if you want to describe the mechanism of converting to `index_type`.  Saying that `index_type` is convertible to the begin index type, which is `index_type`, doesn't make sense (or course a type is convertible to itself, but that isn't what you are trying to say).
- The conversion check cannot be performed for a type that isn't a fundamental type or a type Kokkos knows about.  For a UDT, even if you can construct it with an `index_type`, you don't know that it has the same value it started with.

I think it is better to just say we make a best effort at checking the conversion.

Reply:
I agree that it might just be better for us to state that we try to make best effort on this. There weren't really good ways to summarize the general behavior, especially if UDTs are involved, in which case the conversion checks had to be completely skipped.